zstacklib/zstacklib/utils/qga.py

```
@@ -348,40 +389,62 @@ def guest_exec_python(self, file, params=None, output=True, wait=qga_exec_wait_i
-            ret = self.guest_exec_status(pid)
-            if ret['exited']:
-                break
+            if ret and "pid" in ret:
+                pid = ret["pid"]
+            else:
+                raise Exception(
+                    'qga exec zs-tools operate {} config {} failed for vm {}, ret={}'.format(
+                        operate, config, self.vm_uuid, self._dump_qga_result(ret)))
+
+            ret = None
+            for i in range(retry):
```

Ruff (0.15.21)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)